lib/chef/provider/package.rb

```diff
@@ -44,6 +51,8 @@ def whyrun_supported?
       end
 
       def check_resource_semantics!
+        # FIXME: this is not universally true and subclasses are needing to override this and no-ops it.  It should be turned into
+        # another "subclass_directive" and the apt and yum providers should declare that they need this behavior.
```

Comment:
I like this idea. I was reading the dpkg code and was thinking "looks good, looks good, looks good...", until i saw `def check_resource_semantics!` and it made me sad. Don't know why, it just did.